### DISCUSSION TITLE
Unit mismatch in Dynamic Bias PSD calculation units are not matched with equation.

### DISCUSSION THREAD

**Discussion by venkat057** [rodralez/NaveGo#133] [Q&A] [answered]
Hi,

    In conversions/imu_si_errors.m at line 95 and 101 while calculating Dynamic Bias PSD calculation units are not matched with equation.
    imu_si.ab_psd = imu_si.ab_dyn ./ sqrt(imu.ab_corr); % m/s^2/root(Hz)
    --- Actual units will be % m/s^2/root(time) ....
    --- correct Equation will be imu_si.ab_psd = imu_si.ab_dyn .* sqrt(imu.ab_corr); % m/s^2/root(Hz)
    Please check .
    same for gyro bias PSD.

    In kf_update.m why below equation is being used instead of Joseph stabilized version which is correct as per many references. Is there any advantage with it.?

% Step 5, update the a posteriori covariance matrix, Pp
kf.Pp = kf.Pi - kf.K * kf.S * kf.K'; % Eq. 3.10 from Särkkä

**Comment by rodralez** [comment, marked as answer]
Hello,
 
The only problem is that in fact the PSD unit is **√PSD** (root-PSD), not just PSD.

According to Groves, p. 156, root-PSD is:

```
1 mg/√Hz = 9.80665 * 10^-6 m * s^-1.5 
1 °/hr/√Hz = 4.848 * 10^-6 rad * s^-0.5
```

Where,

```
syms Hz s m

Hz = 1/s;
eq1 = m / s^2 / (Hz)^0.5;
eq2 = simplify(eq1)
eq2 =
 
m*(1/s)^(3/2)
 
```
So, 

` m / s^2 / (Hz)^0.5 == m*(1/s)^(3/2) == m*s^-1.5`
 
Same results can be achieved following the previous steps for `1 °/√hr`
